### DISCUSSION TITLE
Multiple handlers in sequence per message

### DISCUSSION THREAD

**Discussion by Dzhoker** [aiogram/aiogram#1550] [Q&A]
In aiogram 2, you could specify once = False to allow multiple handlers to process a single message.
For example. `dispatcher.message_handlers.once = False`.
How can I do such multiple processing in aiogram 3?
For example, one handler reacts to the text "Hi". The second handler reacts to "How". If the user writes. "Hi. How are you" both handlers should work.
If anything, I have simplified the logic as much as possible in the example.
Thank you in advance for your help.

**Comment by 021-projects** [comment]
Hey! The easiest way I found is to raise a SkipHandler exception in the handlers:
```python
from aiogram import Dispatcher
from aiogram.dispatcher.event.bases import SkipHandler

dp = Dispatcher()

@dp.message()
async def on_message(message):
    print("Handler 1")
    raise SkipHandler

@dp.message()
async def on_message_next(message):
    print("Handler 2")
    raise SkipHandler
```

**Reply by Dzhoker1** [reply]
I'll test. Thanks.

**Comment by JrooTJunior** [comment]
This is an anti-pattern and may lead to unexpected behavior.

It is better to use one handler with separate functionality:

```python
from aiogram import Router
from aiogram.types import Message

router = Router()


@router.message()
async def on_message(message: Message):
    await use_case_1(message)
    await use_case_2(message)


async def use_case_1(message: Message):
    print("Handler 1")


async def use_case_2(message: Message):
    print("Handler 2")
```

Or, depending on your use case, it can be used with middleware: https://docs.aiogram.dev/en/v3.22.0/dispatcher/middlewares.html
